src/plugins/discover/public/plugin.tsx

```diff
@@ -273,6 +265,27 @@ export class DiscoverPlugin
 
         // make sure the index pattern list is up to date
         await dataStart.indexPatterns.clearCache();
+        const useNewFieldsApi = !core.uiSettings.get('discover:searchFieldsFromSource');
+
+        this.docViewsRegistry!.addDocView({
+          title: i18n.translate('discover.docViews.json.jsonTitle', {
+            defaultMessage: 'JSON',
+          }),
+          order: 20,
+          component: ({ hit, indexPattern }) => (
+            <SourceViewer
+              docProps={{
+                index: hit._index,
+                id: hit._id,
+                indexPatternId: indexPattern?.id || '',
+                indexPatternService: dataStart.indexPatterns,
+                requestAllFields: useNewFieldsApi,
+              }}
+              hasLineNumbers
+            />
+          ),
+        });
```

Comment:
Adding the JSON Doc View here, will add one more Doc View every time the plugin is mounted, e.g. when you navigate to Dashboard and back, Dashboard and back ... 

<img width="685" alt="Bildschirmfoto 2021-06-14 um 17 56 32" src="https://user-images.githubusercontent.com/463851/121922612-6b7bba00-cd3a-11eb-8d94-ea925a4e45d3.png">